src/state/REPLState.js

```diff
@@ -94,13 +95,54 @@ class REPLState {
   static Decode(encodedState) {
     let jsonState = JSON.parse(encodedState);
     return new REPLState(
-      decodeBase64(jsonState.source),
-      decodeBase64(jsonState.plugin),
-      jsonState.configs.map((configs) => {
+      decodeBase64(jsonState.base64SourceKey),
+      decodeBase64(jsonState.base64PluginKey),
+      jsonState.configIDs.map(configs => {
         return decodeBase64(configs);
       })
     );
   }
+
+  /**
+   * Link gets the sharing the sharing link
+   * for the given REPL state.
+   * @returns {Promise<string>} String URL.
+   */
+  async Link() {
+    const url = `${REPLState.baseURL}/api/v1/blobs/create`;
+    try {
+      const resp = await fetch(url, {
+        method: "POST",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+        body: this.Encode(),
+      });
+      const message = await resp.json();
+      return `${baseLink()}/share/${message?.id}`;
+    } catch (err) {
+      console.error(err);
+      return err;
+    }
+  }
+
+  /**
+   * REPLState.FromID returns a REPLState given a unique identifier.
+   * @param {string} ID
+   * @returns {Promise<REPLState | null>}
+   */
+  static async FromID(ID) {
+    const url = `${REPLState.baseURL}/api/v1/blobs/get-blob/${ID}`;
```

Comment:
Might need to change this from `/get-blob/${ID}` to `/${ID}` after changes requested in [#12](https://github.com/MLH-Fellowship/babel-sandbox-server/pull/12)
See [route](https://github.com/MLH-Fellowship/babel-sandbox-server/pull/12/files#diff-a9a8d893a8463ac387317e02eec99c96R35)